src/stream-analytics/azext_stream_analytics/_params.py

```diff
@@ -0,0 +1,180 @@
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the input.')
+
+    with self.argument_context('stream-analytics input list') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+
+    with self.argument_context('stream-analytics input test') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the input.')
+
+    with self.argument_context('stream-analytics output create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the output.')
+        c.argument('datasource', type=validate_file_or_dict, help='The datasource of the output.')
+        c.argument('serialization', type=validate_file_or_dict, help='The serialization of the output.')
+
+    with self.argument_context('stream-analytics output delete') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the output.')
+
+    with self.argument_context('stream-analytics output show') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the output.')
+
+    with self.argument_context('stream-analytics output list') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+
+    with self.argument_context('stream-analytics output test') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the output.')
+
+    with self.argument_context('stream-analytics transformation create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the transformation.')
+        c.argument('streaming_units', help='The number of streaming units that the streaming job uses.')
+        c.argument('transformation_query', help='The query that will be run in the streaming job. You can learn more about the Stream Analytics Query Language (SAQL) here: https://msdn.microsoft.com/library/azure/dn834998 .')
```

Comment:
query

Reply:
It  will conflict with the global `query` parameter, so add this prefix. There is a similar case for `output` below.

Reply:
parameter name will keep `--transformation-query`